resource/graph_node.go

```diff
@@ -518,11 +556,25 @@ func (w *GraphNode) resourceStatus() Status {
 		resName = w.current.Name()
 	}
 
+	err := w.lastErr
+	logger := w.getLoggerOrGlobal()
+
+	// check invariants between state and error
+	switch {
+	case w.state == NodeStateUnhealthy && w.lastErr == nil:
+		logger.Warnw("an unhealthy node doesn't have an error", "resource", resName)
+	case w.state == NodeStateUnhealthy && w.lastErr != nil:
```

Comment:
Judging by the log, this state check is intended to be "Ready"

Reply:
ugh yes, bad autocomplete :(